packages/amplify-graphql-model-transformer/publish-notification-lambda/src/index.ts

```diff
@@ -0,0 +1,100 @@
+import { LambdaClient, ListLayerVersionsCommand, ListLayerVersionsCommandOutput } from "@aws-sdk/client-lambda";
+import { PublishCommand, SNSClient } from "@aws-sdk/client-sns";
+
+const MAX_ITEMS = 50;
+const { LAYER_NAME, SNS_TOPIC_ARN, SNS_TOPIC_REGION } = process.env;
+
+const SUPPORTED_REGIONS = [
+  'us-east-1',
+  'us-west-2',
+  'us-west-1',
+  'eu-north-1',
+  'ap-south-1',
+  'eu-central-1',
+  'eu-west-1',
+  'sa-east-1',
+  'ap-southeast-1',
+  'ap-northeast-2',
+  'eu-west-2',
+  'us-east-2',
+  'ap-northeast-1',
+  'eu-west-3',
+  'ap-southeast-2',
+  'ca-central-1',
+  'ap-northeast-3',
+];
+
+type LayerConfig = {
+  layerArn: string | undefined;
+  region: string;
+};
+
+const getLatestVersion = async (layerName: string, region: string): Promise<LayerConfig> => {
+  const client = new LambdaClient({ region });
+  const versions = [];
+  let response: ListLayerVersionsCommandOutput | undefined;
+  do {
+    const command = new ListLayerVersionsCommand({
+      LayerName: layerName,
+      MaxItems: MAX_ITEMS,
+      Marker: response?.NextMarker,
+    });
+    // eslint-disable-next-line no-await-in-loop
+    response = await client.send(command);
+    if (response.LayerVersions) {
+      versions.push(...response.LayerVersions);
+    }
+  } while (response?.NextMarker);
+  const result = versions.sort((a, b) => b.Version! - a.Version!)[0];
+  return {
+    layerArn: result.LayerVersionArn,
+    region,
```

Comment:
result is possibly an empty array, right? should layerArn be `result?.LayerVersionArn`?

Reply:
We query only the regions we support and already published layers. So it should have at least one element.